packages/@ember/-internals/metal/lib/cached.ts

```diff
@@ -0,0 +1,113 @@
+// NOTE: copied from: https://github.com/glimmerjs/glimmer.js/pull/358
```

Comment:
Can probably tweak this comment to mention that both `glimmerjs/glimmer.js` and Ember have copies of this code so folks should make fixes/edits to both locations.

Reply:
tweaked 🙃